src/v2/guide/list.md

```diff
@@ -233,6 +233,8 @@ It is recommended to provide a `key` with `v-for` whenever possible, unless the
 
 Since it's a generic mechanism for Vue to identify nodes, the `key` also has other uses that are not specifically tied to `v-for`, as we will see later in the guide.
 
+<p class="tip">Avoid using non-primitive values as key, use string/number value instead.</p>
```

Comment:
```suggestion
<p class="tip">Don't use non-primitive values like objects and arrays as `v-for` keys. Use string or number values instead.</p>
```

I know the message is taken directly from Vue's warning, but for a documentation purpose, it doesn't need to be that brief ;)